### PR TITLE
`invariant_reserves` failure when `take` / `bucketTake`

### PR DESCRIPTION
**Pull request by grandizzy**
## Description

reserves invariant failure

Add logic in invariant test harness to allow reserves to increase up to ~1e-18th of the current inflator. 

## Purpose

Because debt is stores in t0 terms and reserves include current debt, due to roundoff error the reserves will fluctuate up to `1 WAD` times the current inflator value.  This change allows for this expected error in reserves changes, enabling deeper invariant test runs.  In particular, it fixes regression sequences `test_regression_failure_reserves_bucketTake_18precision` and `test_regression_failure_reserves_regularTake_18precision`.

## Impact

Core contracts are untouched by this change, so no impact on size or gas.

## Tasks

- [ ] Changes to protocol contracts are covered by unit tests executed by CI.
- [ ] Protocol contract size limits have not been exceeded.
- [ ] Gas consumption for impacted transactions have been compared with the target branch, and nontrivial changes cited in the _Impact_ section above.
- [ ] Scope labels have been assigned as appropriate.
- [ ] Invariant tests have been manually executed as appropriate for the nature of the change.

### CODE REVIEW

File: tests/forge/invariants/base/handlers/unbounded/UnboundedLiquidationPoolHandler.sol [hunk 1]
```diff
@@ -293,6 +296,8 @@ abstract contract UnboundedLiquidationPoolHandler is BaseHandler {
             if (auctionInfo.auctionPrice != 0 && auctionInfo.auctionPrice < 100) {
                 reservesErrorMargin = (beforeTakeVars.deposit - afterTakeVars.deposit) / auctionInfo.auctionPrice;
             }
+            (uint256 inflator, ) = _pool.inflatorInfo();
```
Comment (ith-harvey) [inline tests/forge/invariants/base/handlers/unbounded/UnboundedLiquidationPoolHandler.sol:302]: Would appreciate a comment above this line explaining its existence: 
`// Resetting reservesErrorMargin to compensate when the pool has a large inflator as round off errors can occur when calculating reserves. `

File: tests/forge/invariants/base/handlers/unbounded/UnboundedLiquidationPoolHandler.sol [hunk 2]
```diff
@@ -196,11 +196,14 @@ abstract contract UnboundedLiquidationPoolHandler is BaseHandler {
                 kickerReward += afterTakeVars.kickerBond - beforeTakeVars.kickerBond;
             }
 
+            (uint256 inflator, ) = _pool.inflatorInfo();
```
Comment (ith-harvey) [inline tests/forge/invariants/base/handlers/unbounded/UnboundedLiquidationPoolHandler.sol:200]: Would appreciate a comment above this line explaining its existence:
`// Resetting reservesErrorMargin to compensate when the pool has a large inflator as round off errors can occur when calculating reserves. `

**Review by ith-harvey — COMMENTED** [review]
Could be missing something but is this accurate? -> https://github.com/ajna-finance/contracts/blob/6da58b4cdd2f75decac124502da0c55f6c70e9d2/tests/INVARIANTS.md?plain=1#L67

also see -> https://github.com/ajna-finance/contracts/blob/dcae802c991caab094af74fe7a47776117d3b882/tests/forge/invariants/base/ReserveInvariants.t.sol#L20-L21

I think the `reservesErrorMargin` value has a **maximum** of 1e13 value not  `1e15`. Suggest invariants.md is updated to match the code accordingly.

File: tests/forge/invariants/base/handlers/unbounded/UnboundedLiquidationPoolHandler.sol (hunk 2 above)
Comment (mattcushman) [reply tests/forge/invariants/base/handlers/unbounded/UnboundedLiquidationPoolHandler.sol:200]: done

**Review by EdNoepel — APPROVED** [review]
Fine with this.  Note I was not able to reproduce the failure at 600 depth.  Also, merge conflict needs resolution.

**Comment by ith-harvey** [comment]
> Fine with this. Note I was not able to reproduce the failure at 600 depth. Also, merge conflict needs resolution.

Fixed. -> https://github.com/ajna-finance/contracts/pull/1050/commits/dcae802c991caab094af74fe7a47776117d3b882

**Comment by ith-harvey** [comment]
> Could be missing something but is this accurate? ->
> 
> https://github.com/ajna-finance/contracts/blob/6da58b4cdd2f75decac124502da0c55f6c70e9d2/tests/INVARIANTS.md?plain=1#L67
> 
> also see ->
> 
> https://github.com/ajna-finance/contracts/blob/dcae802c991caab094af74fe7a47776117d3b882/tests/forge/invariants/base/ReserveInvariants.t.sol#L20-L21
> 
> I think the `reservesErrorMargin` value has a **maximum** of 1e13 value not `1e15`. Suggest invariants.md is updated to match the code accordingly.

Changed `1e15` -> `1e13` to reflect code -> https://github.com/ajna-finance/contracts/pull/1050/commits/e8abacf5faf474aa2e0351a1598664b610c14831

**Review by ith-harvey — APPROVED** [review]
LGTM

**Review by MikeHathaway — APPROVED** [review]
LGTM

**Pull request merged**